code/game/objects/structures/tables_racks.dm

```diff
@@ -335,6 +392,69 @@
 	log_combat(shover, target, "shoved", "onto [src] (table)")
 	return COMSIG_CARBON_SHOVE_HANDLED
 
+/obj/structure/flippedtable
+	name = "flipped table"
+	desc = "A flipped table."
+	icon = 'icons/obj/flipped_tables.dmi'
+	icon_state = "table"
+	anchored = TRUE
+	density = TRUE
+	layer = ABOVE_MOB_LAYER
+	opacity = FALSE
+	///what type of table the flipped table is
+	var/table_type = /obj/structure/table
+
+/obj/structure/flippedtable/Initialize(mapload)
+	. = ..()
+
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_EXIT = PROC_REF(on_exit),
+	)
+
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/structure/flippedtable/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(table_type == /obj/structure/table/glass) //Glass table, jolly ranchers pass
+		if(istype(mover) && (mover.pass_flags & PASSGLASS))
+			return TRUE
```

Comment:
I hate this istype why is it here, also this code smells a bit but you're workin on it IG

Reply:
It's so that laser shots can pass through the flipped glass table. i'll try to remedy this after by making it a condition/status later on